crud-module/index.js

```diff
@@ -0,0 +1,138 @@
+var Promise = require('bluebird'),
+    child_process = require('child_process'),
+    s = require('underscore.string'),
+    generators = require('yeoman-generator'),
+    mkdirp = require('mkdirp'),
+    log = require('../app/log');
+
+var exec = function (cmd) {
+    return new Promise(function (resolve, reject) {
+        child_process.exec(cmd, function (err, res) {
+            if(err) {
+                reject(err);
+            } else {
+                resolve(res);
+            }
+        });
+    });
+};
+
+// Global Variables
+var folderPath;
+
+module.exports = generators.NamedBase.extend({
+
+    init: function() {
+        this.moduleName = s(this.name).slugify().value();
+        this.capitalizedModuleName = s(this.name).capitalize().value();
+    },
+
+    promptForFolder: function () {
+        var done = this.async();
+
+        var prompt = {
+            name: 'folder',
+            message: 'In which folder would you like the module to be generated? This can be changed later.',
+            default: 'modules'
+        };
+
+        this.prompt(prompt, function (props) {
+            folder = props.folder;
+            folderPath = './' + folder + '/' + this.moduleName + '/';
+            done();
+        }.bind(this));
+    },
+
+    createFolders: function() {
+        mkdirp(folderPath);
```

Comment:
These are all async, they should all be collected into an array an promise.all that